tree/ntuple/v7/test/ntuple_emulated.cxx

```diff
@@ -93,3 +94,108 @@ struct Outer {
    reader = RNTupleReader::Open(cmOpts, "ntpl", fileGuard.GetPath());
    reader->LoadEntry(0);
 }
+
+TEST(RNTupleEmulated, EmulatedFields_Vecs)
+{
+   // Write a user-defined class to RNTuple, then "forget" about it and try to load it as an emulated record field.
+   // To achieve this "forgetting" we use fork() so that the parent process doesn't know about the class.
+
+   FileRaii fileGuard("test_ntuple_emulated_fields_vecs.root");
+
+   ExecInFork([&] {
+      // The child process writes the file and exits, but the file must be preserved to be read by the parent.
+      fileGuard.PreserveFile();
+
+      ASSERT_TRUE(gInterpreter->Declare(R"(
+struct Inner {
```

Comment:
You should uniquely prefix the `struct` names to prevent any caching in the parent process when reading the file.